tensorflow_addons/optimizers/discriminative_layer_training_test.py

```diff
@@ -0,0 +1,467 @@
+        learning_rate = 0.01
+        model = model_fn()
+        model.compile(loss=loss, optimizer=opt(learning_rate * mult))
+
+        model_lr = model_fn()
+        model_lr.lr_mult = mult
+        d_opt = DiscriminativeLayerOptimizer(
+            opt, model_lr, verbose=False, learning_rate=learning_rate
+        )
+        model_lr.compile(loss=loss, optimizer=d_opt)
+
+        self._assert_training_losses_are_close(model, model_lr)
+
+    @test_utils.run_in_graph_and_eager_modes
+    def _test_sub_layers_keep_lr_mult(self, model_fn, loss, opt):
+        """Confirm that model trains with lower lr on specific layer,
+        while a different lr_mult is applied everywhere else.
+        Also confirms that sub layers with an lr mult do not get overridden.
+        """
+
+        learning_rate = 0.01
+        model_lr = model_fn()
+
+        # We set model to lrmult 0 and layer one to lrmult 5.
+        # If layer one is trainable, then the loss should decrease.
+        model_lr.lr_mult = 0.00
+        model_lr.layers[-1].lr_mult = 3
+
+        d_opt = DiscriminativeLayerOptimizer(
+            opt, model_lr, verbose=False, learning_rate=learning_rate
+        )
+        model_lr.compile(loss=loss, optimizer=d_opt)
+
+        loss_values = get_losses(_get_train_results(model_lr, epochs=5))
+        self.assertLess(loss_values[-1], loss_values[0])
+
+    @test_utils.run_in_graph_and_eager_modes
+    def _test_variables_get_assigned(self, model_fn, loss, opt):
+        """Confirm that variables do get an lr_mult attribute and that they get the correct one.
+        """
+        learning_rate = 0.01
+        model_lr = model_fn()
+
+        # set lr mults.
+        model_lr.layers[0].lr_mult = 0.3
+        model_lr.layers[0].layers[-1].lr_mult = 0.1
+        model_lr.layers[-1].lr_mult = 0.5
+
+        d_opt = DiscriminativeLayerOptimizer(
+            opt, model_lr, verbose=False, learning_rate=learning_rate
+        )
+        model_lr.compile(loss=loss, optimizer=d_opt)
+
+        # We expect trainable vars at 0.3 to be reduced by the amount at 0.1.
+        # This tests that the 0.3 lr mult does not override the 0.1 lr mult.
+        self.assertEqual(
+            len(model_lr.layers[0].trainable_variables)
+            - len(model_lr.layers[0].layers[-1].trainable_variables),
+            len([var for var in model_lr.trainable_variables if var.lr_mult == 0.3]),
+        )
+
+        # We expect trainable vars of model with lr_mult 0.1 to equal trainable vars of that layer.
+        self.assertEqual(
+            len(model_lr.layers[0].layers[-1].trainable_variables),
+            len([var for var in model_lr.trainable_variables if var.lr_mult == 0.1]),
+        )
+
+        # Same logic as above.
+        self.assertEqual(
+            len(model_lr.layers[-1].trainable_variables),
+            len([var for var in model_lr.trainable_variables if var.lr_mult == 0.5]),
+        )
+
+    @test_utils.run_in_graph_and_eager_modes
```

Comment:
For testing, we're moving away from this decorator and we're testing in a eager-first setting. This should make your tests simpler. See #1328 . Feel free to ask if you need any help.

Reply:
This may take some time on my end. I'll let you know if I encounter any roadblocks

Reply:
If you wish I can do it. I did it many time so it's quite fast for me to do it. I can push commits directly to your branch.

Reply:
Is the change just removing the @test_utils wrapper? If so, I can do that.

Reply:
It's more complicated. Ideally, we don't use `tf.test.TestCase` anymore.